com.vrcfury.vrcfury/SPS/sps_main.cginc

```diff
@@ -58,18 +58,34 @@ void sps_apply_real(inout float3 vertex, inout float3 normal, uint vertexId, ino
 			applyLerp = min(applyLerp, 1-entranceAngleTooSharp);
 		}
 		
+		const float hiltedSphereRadius = 0.3;
 		if (!isRing) {
 			// Uncancel if hilted in a hole
-			const float hiltedSphereRadius = 0.3;
 			const float inSphere = orfDistance > worldLength*hiltedSphereRadius ? 0 : 1;
 			//const float hilted = min(isBehind, inSphere);
 			//shrinkLerp = hilted;
 			const float hilted = inSphere;
 			applyLerp = max(applyLerp, hilted);
 		} else {
-			// Cancel if ring is near or behind base
-			const float isBehind = rootPos.z > 0 ? 0 : 1;
-			applyLerp = min(applyLerp, 1-isBehind);
+			// Align to axis if hilted in a ring
+			// Prevent orifice from getting too close to plug as that can introduce excessive roll
+			const float minDistance = worldLength * 0.2;
+			const float lerpDistance = minDistance;
+			
+			// Consider plane aligned to orifice
+			const float orfPerpDistance = dot(sps_normalize(-frontNormal), rootPos);
+			float shiftLerp = (orfPerpDistance > minDistance) | (orfPerpDistance < -worldLength * hiltedSphereRadius) ? 0 : 1;
+			
+			const float shiftAmount = (minDistance - orfPerpDistance);
+			const float3 shiftedRoot = rootPos - sps_normalize(frontNormal) * shiftAmount;
+			
+			// Lerp out as ring extends away from plug
+			// https://stackoverflow.com/a/52471226
+			const float orfPlaneDistance = length(shiftedRoot + frontNormal * dot(shiftedRoot, -frontNormal) / dot(frontNormal, frontNormal));
+			shiftLerp *= sps_saturated_map(orfPlaneDistance, minDistance + lerpDistance, minDistance);
+
+			rootPos = lerp(rootPos, shiftedRoot, shiftLerp);
+			orfDistance = length(rootPos);
 		}
```

Comment:
What's the ultimate goal? DPS's proximity solver is... not great, so I'm reluctant to take much inspiration from it at all. Much of DPS's logic feels very much like a "throw things at the wall and see what works", while I think we can do better by defining how the behavior should actually work.

Reply:
Are you referring to this section of code? or the whole PR in general?

Reply:
The ultimate goal was to avoid the sudden rotations in the plug under lateral movements as it gets close to the socket

Ideally I think this should be handled directly in the solver algorithm, however, I was unable to do so.